### PR TITLE
[nfc][Driver] Remove {{(.exe)?}} from sanitizer test

### PR DESCRIPTION
**Pull request by vitalybuka**
These are not Windows tests

### CODE REVIEW

File: clang/test/Driver/sanitizer-ld.c [hunk 1]
```diff
@@ -8,7 +8,7 @@
 // RUN:     --sysroot=%S/Inputs/basic_linux_tree \
 // RUN:   | %{filecheck} --check-prefix=CHECK-ASAN-LINUX
 //
-// CHECK-ASAN-LINUX: "{{(.*[^-.0-9A-Z_a-z])?}}ld{{(.exe)?}}"
+// CHECK-ASAN-LINUX: "{{(.*[^-.0-9A-Z_a-z])?}}ld"
```
Comment (carlocab) [inline clang/test/Driver/sanitizer-ld.c:11]: Is this not needed if you are running this test on Windows?

Comment (vitalybuka) [reply clang/test/Driver/sanitizer-ld.c:11]: I believe `--target=i386-unknown-linux` still make it no-exe even on WIndows.

Comment (vitalybuka) [reply clang/test/Driver/sanitizer-ld.c:11]: Looks like all windows bots are OK https://lab.llvm.org/buildbot/#/builders/193/builds/4341

Comment (vitalybuka) [reply clang/test/Driver/sanitizer-ld.c:11]: It passes on some, and fail on others https://buildkite.com/llvm-project/github-pull-requests/builds/132244

**Pull request merged**